backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/FunctionInlining.kt

```diff
@@ -376,40 +373,40 @@ private class Inliner(val globalSubstituteMap: MutableMap<DeclarationDescriptor,
 
         val parametersWithDefaultToArgument = mutableListOf<ParameterToArgument>()
         irFunction.valueParameters.forEach { parameter ->                                   // Iterate value parameters.
-            val parameterDescriptor = parameter.descriptor as ValueParameterDescriptor
-            val argument = valueArguments[parameterDescriptor.index]           // Get appropriate argument from call site.
+            val argument = valueArguments[parameter.index]                                  // Get appropriate argument from call site.
             when {
                 argument != null -> {                                                       // Argument is good enough.
                     parameterToArgument += ParameterToArgument(                             // Associate current parameter with the argument.
-                        parameterDescriptor = parameterDescriptor,
+                        parameter = parameter,
                         argumentExpression  = argument
                     )
                 }
 
-                parameterDescriptor.hasDefaultValue() -> {                                  // There is no argument - try default value.
-                    val defaultArgument = irFunction.getDefault(parameterDescriptor)!!
+                // After ExpectDeclarationsRemoving pass default values from expect declarations
+                // are represented correctly in IR.
+                parameter.defaultValue != null -> {  // There is no argument - try default value.
                     parametersWithDefaultToArgument += ParameterToArgument(
-                        parameterDescriptor = parameterDescriptor,
-                        argumentExpression  = defaultArgument.expression
+                        parameter = parameter,
+                        argumentExpression  = parameter.defaultValue!!.expression
                     )
                 }
 
-                parameterDescriptor.varargElementType != null -> {
+                parameter.varargElementType != null -> {
                     val emptyArray = IrVarargImpl(
                         startOffset       = irCall.startOffset,
                         endOffset         = irCall.endOffset,
                         type              = parameter.type,
                         varargElementType = parameter.varargElementType!!
                     )
                     parameterToArgument += ParameterToArgument(
-                        parameterDescriptor = parameterDescriptor,
+                        parameter = parameter,
                         argumentExpression  = emptyArray
                     )
                 }
 
                 else -> {
-                    val message = "Incomplete expression: call to $functionDescriptor " +
-                        "has no argument at index ${parameterDescriptor.index}"
+                    val message = "Incomplete expression: call to $irFunction " +
```

Comment:
`IrFunction.toString()` is useless, so it is better to include `functionDescriptor` into error message.